src/closure-types.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+// TODO Object<foo>, Object<foo, bar>
+// TODO Record types
+// TODO Optional param =
+// TODO Function variadic parameters
+//
+// Useful resources for working on this package:
+// https://eslint.org/doctrine/demo/
+// https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System
+
+import * as doctrine from 'doctrine';
+
+/**
+ * Convert from a type annotation in Closure syntax to TypeScript syntax (e.g
+ * `Array<string>` => `string[]|null`).
+ */
+export function closureTypeToTypeScript(closureType: string): string {
+  // Doctrine parses JSDoc annotations, and doesn't have an API purely for type
+  // expressions, so we'll wrap it in a JSDoc annotation and extract the part
+  // of the AST we care about.
+  const annotationAst = doctrine.parse('@return {' + closureType + '}');
+  const typeAst = annotationAst.tags[0].type as doctrine.Type;
+  return serialize(typeAst);
+}
+
+function serialize(node: doctrine.Type): string {
```

Comment:
Add a comment and maybe add toTS to the name?

Reply:
Done.